node/src/actors/chain_manager/mod.rs

```diff
@@ -1015,6 +1016,43 @@ impl ChainManager {
                 chain_info.highest_block_checkpoint = beacon;
                 chain_info.highest_vrf_output = vrf_input;
 
+                let miner_pkh = block.block_header.proof.proof.pkh();
+
+                // Reset the coin age of the miner for all staked coins
+                if get_protocol_version(Some(block_epoch)) == ProtocolVersion::V2_0 {
+                    let _ = stakes.reset_age(miner_pkh, Capability::Mining, current_epoch);
+
+                    let epoch_constants = self.epoch_constants.unwrap();
+                    let utxo_diff =
+                        UtxoDiff::new(&self.chain_state.unspent_outputs_pool, block_epoch);
+
+                    let mut transaction_fees = 0;
+                    for vt_tx in &block.txns.value_transfer_txns {
+                        transaction_fees +=
+                            vt_transaction_fee(vt_tx, &utxo_diff, current_epoch, epoch_constants)
+                                .unwrap_or_default();
+                    }
+                    for dr_tx in &block.txns.data_request_txns {
+                        transaction_fees +=
+                            dr_transaction_fee(dr_tx, &utxo_diff, current_epoch, epoch_constants)
+                                .unwrap_or_default();
+                    }
+                    for st_tx in &block.txns.stake_txns {
+                        transaction_fees +=
+                            st_transaction_fee(st_tx, &utxo_diff, current_epoch, epoch_constants)
+                                .unwrap_or_default();
+                    }
+                    for ut_tx in &block.txns.unstake_txns {
+                        transaction_fees += ut_transaction_fee(ut_tx).unwrap_or_default();
+                    }
+
+                    let _ = stakes.add_reward(
+                        miner_pkh,
+                        Wit::from(50_000_000_000) + Wit::from(transaction_fees),
```

Comment:
Afaik, 50 WIT as block reward is not yet decided. Perhaps this proposal of reducing block time and reward should be part a dedicated PR, not mixed up with a potential fix or improvement in the mining elegibility algorithm.

Reply:
Of course, I fully agree, but I had to put something there as a place holder.